railties/lib/rails/tasks/routes.rake

```diff
@@ -1,7 +1,33 @@
-desc 'Print out all defined routes in match order, with names. Target specific controller with CONTROLLER=x.'
+require 'active_support/deprecation'
+require 'active_support/core_ext/string/strip' # for strip_heredoc
+require 'optparse'
+
+desc 'Print out all defined routes in match order, with names. Target specific controller with --controller option - or its -c shorthand.'
 task routes: :environment do
   all_routes = Rails.application.routes.routes
   require 'action_dispatch/routing/inspector'
   inspector = ActionDispatch::Routing::RoutesInspector.new(all_routes)
-  puts inspector.format(ActionDispatch::Routing::ConsoleFormatter.new, ENV['CONTROLLER'])
+
+  if ENV['CONTROLLER']
+    ActiveSupport::Deprecation.warn <<-eow.strip_heredoc
+      Passing `CONTROLLER` to `bin/rake routes` is deprecated and will be removed in Rails 5.1.
+      Please use `bin/rake routes -c #{ENV['CONTROLLER']}` instead.
```

Comment:
@kaspth FWIW, I'm really not a fan of including user-supplied values in deprecation messages. Even less so if we fail to quote them safely :neutral_face: